spring-cloud-kubernetes-fabric8-loadbalancer/src/main/java/org/springframework/cloud/kubernetes/fabric8/loadbalancer/Fabric8ServicesListSupplier.java

```diff
@@ -38,30 +41,45 @@
  */
 public class Fabric8ServicesListSupplier extends KubernetesServicesListSupplier<Service> {
 
+	private static final LogAccessor LOG = new LogAccessor(LogFactory.getLog(Fabric8ServicesListSupplier.class));
+
 	private final KubernetesClient kubernetesClient;
 
+	private final KubernetesNamespaceProvider namespaceProvider;
+
 	Fabric8ServicesListSupplier(Environment environment, KubernetesClient kubernetesClient,
 			Fabric8ServiceInstanceMapper mapper, KubernetesDiscoveryProperties discoveryProperties) {
 		super(environment, mapper, discoveryProperties);
 		this.kubernetesClient = kubernetesClient;
+		namespaceProvider = new KubernetesNamespaceProvider(environment);
 	}
 
 	@Override
 	public Flux<List<ServiceInstance>> get() {
 		List<ServiceInstance> result = new ArrayList<>();
+		String serviceName = getServiceId();
+		LOG.debug(() -> "serviceID : " + serviceName);
+
 		if (discoveryProperties.allNamespaces()) {
+			LOG.debug(() -> "discovering services in all namespaces");
 			List<Service> services = kubernetesClient.services().inAnyNamespace()
-					.withField("metadata.name", getServiceId()).list().getItems();
+					.withField("metadata.name", serviceName).list().getItems();
 			services.forEach(service -> result.add(mapper.map(service)));
 		}
 		else {
-			Service service = StringUtils.hasText(kubernetesClient.getNamespace()) ? kubernetesClient.services()
-					.inNamespace(kubernetesClient.getNamespace()).withName(getServiceId()).get()
-					: kubernetesClient.services().withName(getServiceId()).get();
+			String namespace = Fabric8Utils.getApplicationNamespace(kubernetesClient, null, "loadbalancer-service",
```

Comment:
Doesn't this change the functionality a bit?  Right now if we don't find the service in the namespace we look for it across all namespaces.  With this change we will just return an empty Flux if we don't find it.

Reply:
Are you saying it is exactly the same because before the `KubernetesClient` that was passed into the class already had the namespace set so when calling `.inNamespace` it didn't make a difference in the services returned?

Reply:
I think I understand now, when you specify the namespace and it does not find the service it will also return the service if it exists in the default namespace?